### PR TITLE
Left-navigation

### PR DESCRIPTION
**Pull request by di**
@pizzapanther Mind taking a look at this WIP and letting me know if it's close to what you were imagining?

Still needs:
 - [x] breakpoints to hide on low-res
 - [x] breakpoints to hide the `SideNav` on high-res
 - [x] figure out why the links are actually working

### CODE REVIEW

**Comment by pizzapanther** [comment]
Definitely right direction. I was thinking of just making it position: fixed or sticky instead of "flex"ing it. Just a couple notes.

1. made the `side-nav` LINKS exported so we can import them into this component and just have one place to edit the links.
2. CSS in `frontend.html` is just for the splash screen before Vue loads except for the background image which had some complexities. So that CSS should probably go in `pytx.less`.
3. Also feel free to make some breakpoints and put them in say a `frontend/less/media.less`. Just haven't gotten to it yet but i know we'll need them at some point.

**Comment by pizzapanther** [comment]
Oh yeah i probably should have mentioned this before:

Format python code: `python run_yapf.py`

Haven't set up the JS formatter yet wasn't expecting any help 🥇 ===> @di 

Let know if you have any questions on the setup, it's a bit custom but I don't think too hard to understand.

**Comment by pizzapanther** [comment]
one more i just saw: we can definitely use `<router-link>` here. I only did `@click="goto()"` for side nav because I wanted it to close after a click. Couldn't find better way to do that.

File: less/media.less [hunk 1]
```diff
@@ -0,0 +1,2 @@
+@for-phone: ~"screen and (max-width: 44.9375rem)"; // < 720
```
Comment (pizzapanther) [inline less/media.less:1]: just curious about using rem's for breakpoints, never used it before what's the advantage of it?

File: frontend.html [hunk 2]
```diff
@@ -74,18 +73,18 @@
         </md-button>
       </div>
       <div id="content">
-        <tab-nav></tab-nav>
+        <left-nav></left-nav>
```
Comment (pizzapanther) [inline frontend.html]: still need the tab nav here too, tab-nav is probably better called sub-nav

File: app/widgets/top-bar.less [hunk 3]
```diff
@@ -33,4 +34,10 @@
       flex: 1;
     }
   }
+
+  .hamburger {
```
Comment (pizzapanther) [inline app/widgets/top-bar.less]: i was going to leave the hamburger even on desktop, kind of weird but I'm seeing this done in a few places and i think it's not a bad idea

File: package.json [hunk 4]
```diff
@@ -5,7 +5,8 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "postinstall": "gulp"
+    "postinstall": "gulp",
+    "prettier": "prettier --write 'frontend/app/**/*.js' && prettier --write 'frontend/**/*.less'"
```
Comment (pizzapanther) [inline package.json:9]: like this, made a bash script on master but this is better

**Review by pizzapanther — COMMENTED** [review]
I'm going to put the tab nav back and change the menu to position: fixed and then merge.

File: less/media.less (hunk 1 above)
Comment (di) [reply less/media.less:1]: I don't really do enough CSS to pretend like I'm an authority on the matter, but my understanding is that this avoids weird edge cases, like when the user has the browser zoomed in.

https://cloudfour.com/thinks/the-ems-have-it-proportional-media-queries-ftw/ seems to discuss it pretty well in depth.

File: app/widgets/top-bar.less (hunk 3 above)
Comment (di) [reply app/widgets/top-bar.less]: I figured it was probably redundant if the left-nav is visible but it doesn't hurt.

File: frontend.html (hunk 2 above)
Comment (di) [reply frontend.html]: Hmm, that might mess up the flexing.

File: package.json (hunk 4 above)
Comment (di) [reply package.json:9]: Oops! Didn't see the script.

File: app/widgets/left-nav.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import { NAV_LINKS } from "../data";
+
+var LeftNav = Vue.component("left-nav", {
+  template: "#tpl-widgets-left-nav",
+  data() {
+    return {
+      links: NAV_LINKS
+    };
+  },
+  methods: {
```
Comment (pizzapanther) [inline app/widgets/left-nav.js:12]: looks like it did need the `goto` because pages are not scrolling correctly. Need to fix that with the Vue Router.

**Comment by di** [comment]
@pizzapanther Not sure what happened but the home page doesn't look great on mobile anymore, but the rest seems OK:

| screenshot_20170906-211421 | 
screenshot_20170906-211731 |
| - | - |

**Comment by pizzapanther** [comment]
don't know why i did it but i had the home page width set to 960px: https://github.com/pytexas/PyTexas2017/commit/869b9772e264f78a244fc0b0399b2c30801b9c4f

**Pull request merged**